src/stores/PathDetailsStore.ts

```diff
@@ -0,0 +1,54 @@
+import Store from '@/stores/Store'
+import { Action } from '@/stores/Dispatcher'
+import { PathDetailsElevationSelected, PathDetailsHover, PathDetailsRangeSelected } from '@/actions/Actions'
+import { Coordinate } from '@/stores/QueryStore'
+import { Bbox } from '@/api/graphhopper'
+
+export interface PathDetailsPoint {
+    point: Coordinate
+    elevation: string
+    description: string
+}
+
+export interface PathDetailsStoreState {
+    pathDetailsPoint: PathDetailsPoint | null
+    pathDetailBbox?: Bbox
+    pathDetailsHighlightedSegments: Coordinate[][]
+}
+
+export default class PathDetailsStore extends Store<PathDetailsStoreState> {
+    constructor() {
+        super()
+    }
+
+    protected getInitialState(): PathDetailsStoreState {
+        return {
```

Comment:
I find it nice to not have null or undefined values. Other than Java Typescript somewhat forces us to never have undefined values since it is necessary to separately declare that type. Not having undefined and null removes a lot of places where one has to test for presence.

If you can make it work maybe it is possible to supply an empty but valid state initially and design the components so that they always display the right thing also with an initial state.

Reply:
Yes, I also noticed this. Using types like `<type> | null` does seem a bit strange indeed. Then again it kind of expresses exactly what I want. `pathDetailsPoint` is the point on the map that corresponds to the mouse hover position in the height graph. When the mouse is not positioned above the heightgraph there is no such point and I use `null` to cover this state. What would be the alternative? We could use some special point value like `[600,600]` to represent 'no point selected', but so far I do not see why this would be better (and I think its worse tbh). Sure we would not need the extra `| null` in the typing but then we need to do another comparison like `point !== [600,600]` elsewhere. In this specific example there is only one place where we check if the point is `null` and this is the place where we have to decide whether there should or should not be a marker on the map. I think it would be the same if we had to check for some special value. 

But generally I agree that thinking about null or undefined values can make the code worse and my default assumption is also that nothing is null or undefined (same in Java btw) unless cases like this where there is a notion of a value possibly 'missing. Hm, another possibility could be adding something like `pointSelected: boolean` to the state?

Reply:
I see that it makes sense in this case. I also agree that the special value approach would be worse.